packages/keyring-controller/src/KeyringController.ts

```diff
@@ -658,18 +674,27 @@ export class KeyringController extends BaseController<
    * Adds a new account to the default (first) HD seed phrase keyring.
    *
    * @param accountCount - Number of accounts before adding a new one, used to
+   * @param keyringId - The id of the keyring to add the account to.
    * make the method idempotent.
    * @returns Promise resolving to the added account address.
    */
-  async addNewAccount(accountCount?: number): Promise<string> {
+  async addNewAccount(
+    accountCount?: number,
+    keyringId?: string,
+  ): Promise<string> {
```

Comment:
There seems to be no restriction now on what kind of Keyring can be executed this operation on. Though, this would not make sense for almost all keyrings we have besides the HD one.

What are the benefits of using an ID, instead of a type and an index?